src/discordcr/client.cr

```diff
@@ -94,18 +94,25 @@ module Discord
       spawn do
         packet = parse_message(message)
 
-        case packet.opcode
-        when OP_HELLO
-          payload = Gateway::HelloPayload.from_json(packet.data)
-          handle_hello(payload.heartbeat_interval)
-        when OP_DISPATCH
-          handle_dispatch(packet.event_type, packet.data)
-        when OP_RECONNECT
-          handle_reconnect
-        when OP_INVALID_SESSION
-          handle_invalid_session
-        else
-          puts "Unsupported message: #{message}"
+        begin
+          case packet.opcode
+          when OP_HELLO
+            payload = Gateway::HelloPayload.from_json(packet.data)
+            handle_hello(payload.heartbeat_interval)
+          when OP_DISPATCH
+            handle_dispatch(packet.event_type, packet.data)
+          when OP_RECONNECT
+            handle_reconnect
+          when OP_INVALID_SESSION
+            handle_invalid_session
+          else
+            puts "Unsupported message: #{message}"
+          end
+        rescue ex
+          puts "An exception occurred during message parsing!"
```

Comment:
Is this really just during message parsing? If a handler raises *any* exception won't it filter up to here? This includes user handlers and caching exceptions.